common/optimiser/OptimiserVariable.java

```diff
@@ -26,27 +26,60 @@
 
 public abstract class OptimiserVariable<T> {
 
+    protected T initial;
+    protected T solution;
+    protected MPVariable mpVariable;
+
     final String name;
 
     OptimiserVariable(String name) {
         this.name = name;
     }
 
-    public abstract T solutionValue();
+    public T value() {
+        if (hasSolutionValue()) return solutionValue();
+        else return initial();
+    }
 
-    abstract MPVariable mpVariable();
+    public boolean hasSolutionValue() {
+        return solution != null;
+    }
 
-    abstract void recordValue();
+    public T solutionValue() {
+        assert hasSolutionValue();
+        return solution;
+    }
+
+    public void setInitial(T initial) {
+        this.initial = initial;
+    }
+
+    public void clearInitial() {
+        initial = null;
+    }
+
+    public boolean hasInitial() {
+        return initial != null;
+    }
+
+    public T initial() {
+        assert hasInitial();
+        return initial;
+    }
 
-    public abstract void clearInitial();
+    public abstract double hint();
```

Comment:
We now have `initial()` to retrieve the native initial value, and `hint()` to convert it to a `double` for MPSolver.